drivers/mailbox/bcm2836/device.c

```diff
@@ -430,6 +430,190 @@ Return Value:
     NTSTATUS value
 
 --*/
+
+extern WCHAR macAddrStrGlobal[13];
+
+NTSTATUS RpiqSetNdisMacAddress()
+{
+    NTSTATUS status = STATUS_SUCCESS;
+    UNICODE_STRING KeyPrefix = RTL_CONSTANT_STRING (
+        L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\Class\\{4d36e972-e325-11ce-bfc1-08002be10318}\\");
+    UNICODE_STRING DriverDesc = RTL_CONSTANT_STRING ( L"DriverDesc" );
+    UNICODE_STRING Desc;
+    UNICODE_STRING DescLAN7800 = RTL_CONSTANT_STRING ( L"LAN7800" );
+    UNICODE_STRING NetworkAddress = RTL_CONSTANT_STRING ( L"NetworkAddress" );
+    UNICODE_STRING PropertyChangeStatus = RTL_CONSTANT_STRING ( L"PropertyChangeStatus" );
+    UNICODE_STRING KeyName;
+    WCHAR KeyInstance[] = L"0000";
+    ULONG Instance = 0;
+    static BOOLEAN MacSet = 0;
+    OBJECT_ATTRIBUTES ObjectAttributes;  
+    HANDLE BaseKey = NULL;
+    HANDLE SubKey = NULL;
+    PKEY_VALUE_FULL_INFORMATION infoBuffer = NULL;
+    ULONG infoBufferLength = 0;
+    ULONG RequireBufferLength = 0;
+    DWORD value;
+
+    if(MacSet) {
+    return status;
+        }
+
+    MacSet = TRUE;
+
+    InitializeObjectAttributes (&ObjectAttributes,
+                                (PUNICODE_STRING)&KeyPrefix,
+                                OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+                                NULL,
+                                NULL
+                                );
+
+    status = ZwOpenKey (&BaseKey,
+                        KEY_READ | KEY_WRITE,
+                        &ObjectAttributes);
+    if (!NT_SUCCESS(status)){
+        BaseKey = NULL;
+        goto End;
+    }
+    
+    KeyName.Length = 8;
+    KeyName.MaximumLength = 8;
+    KeyName.Buffer = KeyInstance;
+
+    while(STATUS_OBJECT_NAME_NOT_FOUND != status)
```

Comment:
Maybe add a comment with the high level overview of what the loop below achieves and how.

i.e. iterates through all enumerated ndis adapters looking for the LAN7800 instance and sets its MAC address if needed.